File: l0019_remove_nth_node_from_end/remove_nth_node_from_end.go
package leetcode0019

type ListNode struct {
	Val  int
	Next *ListNode
}

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	dummy := new(ListNode)
	dummy.Next = head
	slow := dummy
	fast := dummy
	for i := 0; i < n; i++ {
		fast = fast.Next
	}
	for fast.Next != nil {
		fast = fast.Next
		slow = slow.Next
	}
	slow.Next = slow.Next.Next
	return dummy.Next
}

File: l0019_remove_nth_node_from_end/remove_nth_node_from_end.c
#include <stdlib.h>

struct ListNode {
	int var;
	struct ListNode *next;
};

struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
	struct ListNode *dummy = malloc(sizeof(struct ListNode));
	dummy->next = head;
	struct ListNode *fast = dummy;
	struct ListNode *slow = dummy;
	for (int i = 0; i < n; i++) {
		fast = fast->next;
	}
	while (fast->next) {
		fast = fast->next;
		slow = slow->next;
	}
	slow->next = slow->next->next;
	return dummy->next;
}